### PR TITLE
Post Registration Data

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/EUI-8814

### Checklist

- [ x ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ x ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/register-org/services/register-org.service.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { SessionStorageService } from '../../shared/services/session-storage.service';
-import { RegistrationData } from '../models/registration-data.model';
+import { Observable } from 'rxjs';
```
Comment (connorpgpmcelroy) [inline src/register-org/services/register-org.service.ts:4]: import ordering - minor issue

File: src/register-org/components/individual-registered-with-regulator/individual-registered-with-regulator.component.spec.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { Router } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
 import { RegistrationData } from '../../models/registration-data.model';
 import { IndividualRegisteredWithRegulatorComponent } from './individual-registered-with-regulator.component';
+import { RegisterOrgService } from '../../services/register-org.service';
```
Comment (connorpgpmcelroy) [inline src/register-org/components/individual-registered-with-regulator/individual-registered-with-regulator.component.spec.ts:6]: ordering

File: src/register-org/components/company-house-details/company-house-details.component.spec.ts [hunk 3]
```diff
@@ -3,17 +3,29 @@ import { Router } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
 import { CompanyHouseDetailsMessage } from '../../../register-org/models';
 import { CompanyHouseDetailsComponent } from './company-house-details.component';
+import { RegisterOrgService } from '../../services/register-org.service';
```
Comment (connorpgpmcelroy) [inline src/register-org/components/company-house-details/company-house-details.component.spec.ts:6]: ordering

File: src/register-org/components/check-your-answers/regulator-list/regulator-list.component.spec.ts [hunk 4]
```diff
@@ -1,12 +1,22 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { RegulatorType } from 'src/register-org/models';
 import { RegulatorListComponent } from './regulator-list.component';
+import { RegisterOrgService } from '../../../services/register-org.service';
```
Comment (connorpgpmcelroy) [inline src/register-org/components/check-your-answers/regulator-list/regulator-list.component.spec.ts:4]: ordering

File: src/register-org/components/check-your-answers/check-your-answers.component.ts [hunk 5]
```diff
@@ -1,10 +1,11 @@
 import { Component, OnInit } from '@angular/core';
-import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AbstractControl, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
 import { Router } from '@angular/router';
 import { RegisterComponent } from '../../../register-org/containers';
 import { ORGANISATION_SERVICES } from '../../constants/register-org-constants';
 import { RegulatorType, RegulatoryType } from '../../models';
 import { RegisterOrgService } from '../../services/register-org.service';
+import { HttpErrorResponse } from '@angular/common/http';
```
Comment (connorpgpmcelroy) [inline src/register-org/components/check-your-answers/check-your-answers.component.ts:8]: ordering

File: src/register-org/components/check-your-answers/check-your-answers.component.ts [hunk 6]
```diff
@@ -38,9 +42,54 @@ export class CheckYourAnswersComponent extends RegisterComponent implements OnIn
...
+    };
+  }
+
   public onBack(): void {
     this.registrationData.hasIndividualRegisteredWithRegulator
       ? this.router.navigate([this.registerOrgService.REGISTER_ORG_NEW_ROUTE, 'individual-registered-with-regulator-details', true])
       : this.router.navigate([this.registerOrgService.REGISTER_ORG_NEW_ROUTE, 'individual-registered-with-regulator', true]);
   }
+
+  public onSubmitData() {
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.ts:61]: @johnbenjamin-hmcts - missing function return type

File: api/openRoutes.ts [hunk 7]
```diff
@@ -8,6 +8,7 @@ import getLovRefDataRouter from './prd/lov';
 // TODO: rename from prdRouter
 import getappInsightsInstrumentationKey from './monitoring-tools';
 import prdRouter from './register-org';
+import registerOrgRouter from './registerOrganisation';
 import addressRouter from './addresses';
```
Comment (connorpgpmcelroy) [inline api/openRoutes.ts:12]: ordering

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Import ordering issues

File: api/registerOrganisation/index.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { Request, Response, Router } from 'express';
+import { RegistrationData } from '../models/registrationData';
+
+export const router = Router({ mergeParams: true });
+
+export async function handleRegisterOrgRoute(req: Request, res: Response) {
```
Comment (Daniel-Lam) [inline api/registerOrganisation/index.ts:6]: @johnbenjamin-hmcts - missing function return type

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - two minor comments on missing return types

File: api/registerOrganisation/index.ts (hunk 8 above)
Comment (udaydenduluri33) [reply api/registerOrganisation/index.ts]: @Daniel-Lam  Done

File: src/register-org/components/check-your-answers/check-your-answers.component.ts (hunk 6 above)
Comment (udaydenduluri33) [reply src/register-org/components/check-your-answers/check-your-answers.component.ts]: @Daniel-Lam  Done

File: api/openRoutes.ts (hunk 7 above)
Comment (connorpgpmcelroy) [reply api/openRoutes.ts:12]: addresses alphabetically before others. However this is openRoutes so actually not necessary

File: src/register-org/components/company-house-details/company-house-details.component.spec.ts (hunk 3 above)
Comment (connorpgpmcelroy) [reply src/register-org/components/company-house-details/company-house-details.component.spec.ts:6]: ../../ should come before ./

File: src/register-org/components/individual-registered-with-regulator/individual-registered-with-regulator.component.spec.ts (hunk 2 above)
Comment (connorpgpmcelroy) [reply src/register-org/components/individual-registered-with-regulator/individual-registered-with-regulator.component.spec.ts:6]: ../../ should come before ./

**Review by Mo-Lala84 — APPROVED** [review]
approved

File: src/register-org/components/check-your-answers/regulator-list/regulator-list.component.spec.ts (hunk 4 above)
Comment (connorpgpmcelroy) [reply src/register-org/components/check-your-answers/regulator-list/regulator-list.component.spec.ts:4]: ../../.. should come before ./ and src mentioned

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**